src/main/kotlin/cacophony/automata/DFAEquivalenceHelper.kt

```diff
@@ -69,15 +65,13 @@ private fun <StateA, StateB, AtomType, ResultType> initializeDistinguishableStat
     return distinguishable
 }
 
-private fun <State, AtomType> getSymbols(productions: Map<Pair<State, AtomType>, State>): Set<AtomType> {
-    return productions.keys.map { it.second }.toSet()
-}
+private fun <State, AtomT> getSymbols(productions: Map<Pair<State, AtomT>, State>): Set<AtomT> = productions.keys.map { it.second }.toSet()
 
-private fun <State, AtomType> invertProductions(
-    dfa: DFA<State, AtomType, *>,
-    symbols: Set<AtomType>,
-): Map<Pair<State?, AtomType>, Set<State?>> {
-    val inverted = mutableMapOf<Pair<State?, AtomType>, MutableSet<State?>>()
+private fun <State, AtomT> invertProductions(
```

Comment:
Why not `StateT`?

Reply:
Yeah, thanks, actually I spoted a few more places, take a look

Reply:
Fine by me. You can merge